.github/workflows/build_and_test.yml

```diff
@@ -126,144 +138,85 @@ jobs:
-
-      - name: "Setup Keys"
+      - name: Load Worker & Client Images
         env:
-          KEY: ${{ secrets.IAS_PRIMARY_KEY }}
-          SPID: ${{ secrets.IAS_SPID }}
-          TLS_CERTIFICATE: ${{ secrets.TLS_WS_SERVER_CERTIFICATE }}
-          TLS_PRIVATE_KEY: ${{ secrets.TLS_WS_SERVER_PRIVATE_KEY }}
+          DOCKER_BUILDKIT: 1
+#        run: |
+#          cat integritee-worker-${{ matrix.mode }}-${{ github.sha }}.tar.gz | docker import - integritee-worker-${{ matrix.mode }}-${{ github.sha }}
+#          cat integritee-cli-client-${{ matrix.mode }}-${{ github.sha }}.tar.gz | docker import - integritee-cli-client-${{ matrix.mode }}-${{ github.sha }}
+#          docker images --all
         run: |
-          cd ${{ env.BIN_DIR }}
-          echo "${{ env.KEY }}" > key.txt
-          echo "${{ env.SPID }}" > spid.txt
-          echo "${{ env.TLS_CERTIFICATE }}" > end.fullchain
-          echo "${{ env.TLS_PRIVATE_KEY }}" > end.rsa
-          chmod 644 end.fullchain
-          chmod 644 end.rsa
-
-      - name: Run local setup
-        # * `set -eo pipefail` is needed to return an error even if piped to `tee`.
-        shell: bash --noprofile --norc -eo pipefail {0}
+          docker image load --input integritee-worker-${{ matrix.mode }}-${{ github.sha }}.tar.gz
+          docker image load --input integritee-cli-client-${{ matrix.mode }}-${{ github.sha }}.tar.gz
+          docker images --all
+
+      - name: Re-name Image Tags
         run: |
-          touch ${{ env.LOG_DIR }}/local-setup.log
-          ./local-setup/launch.py local-setup/github-action-config.json 2>&1 | tee -i ${{ env.LOG_DIR }}/local-setup.log &
-          sleep 150
-
-      - name: ${{ matrix.demo_name }}-${{ matrix.mode }}
-        # * the change the symbolic link which points to the target/release... folder.
-        # * need overwrite default shell to bash to get access to the `source` cmd.
-        shell: bash --noprofile --norc -eo pipefail {0}
+          docker tag integritee-worker-${{ matrix.mode }}-${{ github.sha }} ${{ env.WORKER_IMAGE_TAG }}
+          docker tag integritee-cli-client-${{ matrix.mode }}-${{ github.sha }} ${{ env.CLIENT_IMAGE_TAG }} 
+          docker images --all
+
+      - name: Integration Test ${{ matrix.test }}-${{ matrix.mode }}
         run: |
-          source ./scripts/init_env.sh && ./scripts/${{ matrix.demo_script }}
+          cd docker
+          docker-compose -f docker-compose.yml -f ${{ matrix.demo_name }}.yml up --no-build --exit-code-from ${{ matrix.demo_name }}
 
-      - name: Upload logs
+      - name: Collect Docker Logs
         continue-on-error: true
+        if: always()
+        uses: jwalton/gh-docker-logs@v2.2.0
+        with:
+          #images: '${{ env.WORKER_IMAGE_TAG }},${{ env.CLIENT_IMAGE_TAG }}'
+          tail: all
+          dest: './logs'
+
+      - name: Tar logs
+        if: always()
+        run: tar cvzf ./logs-${{ matrix.test }}-${{ matrix.mode }}.tgz ./logs
```

Comment:
Collect the logs from `docker logs` for all services. We now have log output from the CLI, workers by default. The node logs need to be enabled manually.